internal/btf/core.go

```diff
@@ -424,8 +449,23 @@ func coreCalculateFixup(local Type, localID TypeID, target Type, targetID TypeID
 				return zero, err
 			}
 
-			return fixup(uint32(localSize), uint32(targetSize))
+			return fixupField(uint32(localSize), uint32(targetSize))
+
+		case reloFieldLShiftU64:
+			targetSize, err := Sizeof(targetField.Type)
+			if err != nil {
+				return zero, err
+			}
+			var target uint32
+			if byteOrder == binary.LittleEndian {
+				target = 64 - (targetField.bitfieldOffset + targetField.bitfieldSize - targetField.offset)
+			} else {
+				target = (8-uint32(targetSize))*8 + (targetField.bitfieldOffset - targetField.offset)
+			}
+			return COREFixup{Kind: relo.kind, Local: 0, Target: target, Poison: false, Validate: false}, nil
 
+		case reloFieldRShiftU64:
+			return COREFixup{Kind: relo.kind, Local: 0, Target: 64 - targetField.bitfieldSize, Poison: false, Validate: false}, nil
```

Comment:
In [`libbpf`](https://github.com/libbpf/libbpf/blob/6edaacad4f7c31f08f63205e2b3569b9f946c8cb/src/relo_core.c#L700), `validate` is set to true for right shift:

```C
		*val = 64 - bit_sz;
		if (validate)
			*validate = true; /* right shift is never ambiguous */
		break;
```

But maybe I am too focused on copying what `libbpf` does.

Reply:
it's a different type of validation. I think I'll rename 'validation' to 'checkLocal' or something.